KAFKA-18597 Fix max-buffer-utilization-percent is always 0

LoganZhuZzz opened the pull request:
https://issues.apache.org/jira/browse/KAFKA-18597

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

LoganZhuZzz committed:
KAFKA-18597: Fix max-buffer-utilization-percent is always 0

chia7712 commented:
@LoganZhuZzz This is a bug fix, so could you please add unit test?

LoganZhuZzz commented:
> @LoganZhuZzz This is a bug fix, so could you please add unit test?

I’ll add the unit test later.

chia7712 commented:
@LoganZhuZzz additionally, please take a look at @ijuma's comment (https://github.com/apache/kafka/pull/17390#discussion_r1921578303)

LoganZhuZzz committed:
Merge upstream changes and resolve conflicts

LoganZhuZzz committed:
KAFKA-18597: Fix test issue

LoganZhuZzz committed:
KAFKA-18597: Add unit test

LoganZhuZzz commented:
> @LoganZhuZzz additionally, please take a look at @ijuma's comment ([#17390 (comment)](https://github.com/apache/kafka/pull/17390#discussion_r1921578303))

@chia7712 PTAL

chia7712 commented on core/src/main/scala/kafka/log/LogCleaner.scala:
```diff
@@ -116,14 +116,14 @@ class LogCleaner(initialConfig: CleanerConfig,
   /* a metric to track the maximum utilization of any thread's buffer in the last cleaning */
   metricsGroup.newGauge(MaxBufferUtilizationPercentMetricName,
-    () => maxOverCleanerThreads(_.lastStats.bufferUtilization) * 100)
+    () => (maxOverCleanerThreads(_.lastStats.bufferUtilization) * 100).toInt)
```
Could you please add unit test for other impacted metrics?

chia7712 commented on core/src/test/scala/unit/kafka/log/LogCleanerTest.scala:
```diff
@@ -2070,18 +2071,55 @@ class LogCleanerTest extends Logging {
+  }
+
+  @Test
+  def testMaxBufferUtilizationPercentMetric(): Unit = {
```
It seems this new unit test is similar to `testMaxOverCleanerThreads`, right? If so, could you please remove `testMaxOverCleanerThreads`?

chia7712 reviewed (commented):
@LoganZhuZzz thanks for this patch!

LoganZhuZzz committed:
KAFKA-18597: Add unit tests for additional impacted metrics in LogCleanerTest

- Added testMaxBufferUtilizationPercentMetric to validate maximum buffer utilization metric.
- Added testMaxCleanTimeMetric to verify the maximum clean time metric.
- Added testMaxCompactionDelayMetrics to ensure correctness of maximum compaction delay metric.
